src/data/polynomial/degree/basic.lean

```diff
@@ -194,14 +194,24 @@ begin
   { exact (coeff_eq_zero_of_nat_degree_lt (not_le.1 (λ w, h (nat.lt_one_add_iff.2 w)))).symm, }
 end
 
-lemma as_sum (p : polynomial R) :
+lemma as_sum_range (p : polynomial R) :
   p = ∑ i in range (p.nat_degree + 1), C (p.coeff i) * X^i :=
 begin
   ext n,
   simp only [add_comm, coeff_X_pow, coeff_C_mul, finset.mem_range,
     finset.sum_mul_boole, finset_sum_coeff, ite_le_nat_degree_coeff],
 end
 
+lemma as_sum_support (p : polynomial R) :
+  p = ∑ i in p.support, C (p.coeff i) * X^i :=
+begin
+  ext n,
+  simp only [finset_sum_coeff, coeff_C_mul, coeff_X_pow, finset.sum_mul_boole],
+  split_ifs,
+  { refl },
+  { contrapose h, rw not_not, exact mem_support_iff_coeff_ne_zero.mpr h }
+end
```

Comment:
Can you prove this using `sum_over_range'`?

Reply:
What is the benefit of proving it using `sum_over_range'`, or what is the disadvantage of proving it this way?

Reply:
Really my thinking is that of `sum_over_range'`, `as_sum_support`, and `as_sum_range`, two of them are sufficient to prove the third - there's no need to prove all three from more basic lemmas.

Reply:
`sum_over_rang'` is a weird one, that one is actually using `finsupp.sum`. If we decide to prove `as_sum_support` in terms of something else, I think `as_sum_range` is a better choice.

Reply:
I tried to poke around a bit, can not find an elegant solution to write it in terms of another (

Reply:
```lean
lemma as_sum_support (p : polynomial R) :
  p = ∑ i in p.support, C (p.coeff i) * X^i :=
begin
  change p = p.sum (λ i a, C a * X ^ i),
  rw sum_over_range,
  exact as_sum_range _,
  simp only [forall_const, zero_mul, ring_hom.map_zero],
end
```

Reply:
@shingtaklam1324 Thank you.